### PR TITLE
Add pagination to the home and client list page

### PR DESCRIPTION
**Pull request by Patrick7321**

### CODE REVIEW

File: rim/views.py [hunk 1]
```diff
@@ -12,14 +13,22 @@
 from rim.models import Equipment, Checkout, EquipmentType, Location, Client
 from rim.forms import EquipmentForm
 
-class HomeView(ListView):
+class PaginateView(object):
+    def get_paginate_by(self, queryset):
+        if 'paginate' in self.request.COOKIES:
+            return self.request.COOKIES['paginate']
```
Comment (shaneallgeier) [inline rim/views.py]: You should be doing some additional validation here. What if someone set their cookie to negative numbers or infinity or 1.666 or "asdf"?

Comment (shaneallgeier) [reply rim/views.py]: If it *is* an invalid cookie, [tell the client to delete it](https://docs.djangoproject.com/en/2.1/ref/request-response/#django.http.HttpResponse.delete_cookie).

File: rim/templates/rim/client_list.html [hunk 2]
```diff
@@ -2,6 +2,10 @@
 {% load i18n %}
 {% load static %}
 
+{% block css %}
+    {{ block.super }}
+{% endblock %}
```
[the remark this answers is not in the document]
Comment (Patrick7321) [reply rim/templates/rim/client_list.html]: Kinda pretentious, no?

File: rim/views.py [hunk 3]
```diff
@@ -12,14 +13,36 @@
...
+            self.invalid_per_page = True
+        except KeyError:
+            pass
+        obj_per_page = min(obj_per_page, 1000)
+        obj_per_page = max(obj_per_page, 1)
+        return obj_per_page
+
+    def dispatch(self, *args, **kwargs):
+        response = super().dispatch(*args, **kwargs)
+        if(self.invalid_per_page):
```
Comment (shaneallgeier) [inline rim/views.py]: TOTALLY UNNECESSARY PARENS OMGOMGOG

File: rim/views.py [hunk 4]
```diff
@@ -2,6 +2,7 @@
 
 from django.shortcuts import render
 from django.urls import reverse_lazy
+from django.core.paginator import Paginator
```
Comment (shaneallgeier) [inline rim/views.py]: Unused import

File: rim/views.py [hunk 5]
```diff
@@ -12,14 +13,36 @@
 from rim.models import Equipment, Checkout, EquipmentType, Location, Client
 from rim.forms import EquipmentForm
 
-class HomeView(ListView):
+class PaginateView(object):
```
Comment (shaneallgeier) [inline rim/views.py]: We should probably call this PaginateMixin instead (since it's not really a view)

File: rim/templates/rim/client_list.html [hunk 6]
```diff
@@ -38,4 +38,34 @@ <h1>{% trans 'Client List' %}</h1>
...
+                </div>
+                <div class='small-6 large-6 cell'>
+                    {% if is_paginated %}
+                        <div class='page_nav'>
+                            {% if page_obj.has_previous %}
+                                <a class='page_link' href='?page={{ page_obj.previous_page_number }}'>Previous</a> |
+                            {% endif %}
+                            Page {{ page_obj.number }}
+                            {% if page_obj.has_next %}
+                                 | <a class='page_link' href='?page={{ page_obj.next_page_number }}'>Next</a>
```
Comment (shaneallgeier) [inline rim/templates/rim/client_list.html]: These links need to include the search params, if present

**Pull request merged**